cli/build.go

```diff
@@ -758,40 +771,65 @@ func sha256File(path string) (string, error) {
 
 // TODO: non-linux things too if we ever support non-Linux containers
 // TODO: on non-Linux platforms we shouldn't constantly try to re-download it
-func DownloadYB(ctx context.Context) (string, error) {
+func DownloadYB(ctx context.Context) (_ io.ReadCloser, size int64, _ error) {
 	// Stick with this version, we can track some relatively recent version because
 	// we will just update anyway so it doesn't need to be super-new unless we broke something
-	downloadURL := "https://bin.equinox.io/a/7G9uDXWDjh8/yb-0.0.39-linux-amd64.tar.gz"
-	binarySha256 := "3e21a9c98daa168ea95a5be45d14408c18688b5eea211d7936f6cd013bd23210"
-	cachePath := plumbing.CacheDir()
-	tmpPath := filepath.Join(cachePath, ".yb-tmp")
-	ybPath := filepath.Join(tmpPath, "yb")
-	if err := os.MkdirAll(tmpPath, 0777); err != nil {
-		return "", fmt.Errorf("download yb: %w", err)
-	}
-
-	if plumbing.PathExists(ybPath) {
-		checksum, err := sha256File(ybPath)
-		// Checksum match? We're good to go
-		if err == nil && checksum == binarySha256 {
-			return ybPath, nil
-		}
-
-		log.Infof(ctx, "Local binary checksum mis-match, re-downloading YB")
+	const downloadURL = "https://bin.equinox.io/a/7G9uDXWDjh8/yb-0.0.39-linux-amd64.tar.gz"
+	archivePath, err := plumbing.DownloadFileWithCache(ctx, http.DefaultClient, downloadURL)
+	if err != nil {
+		return nil, 0, fmt.Errorf("download yb: %w", err)
 	}
-
-	// Couldn't tell, check if we need to and download the archive
-	localFile, err := plumbing.DownloadFileWithCache(ctx, http.DefaultClient, downloadURL)
+	archiveFile, err := os.Open(archivePath)
 	if err != nil {
-		return "", fmt.Errorf("download yb: %w", err)
+		return nil, 0, fmt.Errorf("download yb: %w", err)
 	}
-
-	err = archiver.Unarchive(localFile, tmpPath)
+	zr, err := gzip.NewReader(archiveFile)
 	if err != nil {
-		return "", fmt.Errorf("download yb: decompress: %v", err)
+		archiveFile.Close()
+		return nil, 0, fmt.Errorf("download yb: %s: %w", archivePath, err)
+	}
+	archive := tar.NewReader(zr)
+	found := false
+	for {
+		hdr, err := archive.Next()
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			zr.Close()
+			archiveFile.Close()
+			return nil, 0, fmt.Errorf("download yb: %s: %w", archivePath, err)
+		}
+		if hdr.Name == "yb" {
+			found = true
+			size = hdr.Size
+			break
+		}
 	}
+	if !found {
+		zr.Close()
+		archiveFile.Close()
+		return nil, 0, fmt.Errorf("download yb: no 'yb' found in %s", downloadURL)
+	}
+	return multiCloseReader{
+		Reader:  archive,
+		closers: []io.Closer{zr, archiveFile},
+	}, size, nil
```

Comment:
Can this code have the `found` and `size` pieces of state simplified away by moving this `return` up into the loop?
```suggestion
	for {
		hdr, err := archive.Next()
		if errors.Is(err, io.EOF) {
			break
		}
		if err != nil {
			zr.Close()
			archiveFile.Close()
			return nil, 0, fmt.Errorf("download yb: %s: %w", archivePath, err)
		}
		if hdr.Name == "yb" {
			return multiCloseReader{
				Reader:  archive,
				closers: []io.Closer{zr, archiveFile},
			}, hdr.Size, nil
		}
	}
	zr.Close()
	archiveFile.Close()
	return nil, 0, fmt.Errorf("download yb: no 'yb' found in %s", downloadURL)

```